File: conformance/tests/boolean-logic/boolean-logic.runar.py
from runar import SmartContract, Bigint, public, assert_

class BooleanLogic(SmartContract):
    threshold: Bigint

    def __init__(self, threshold: Bigint):
        super().__init__(threshold)
        self.threshold = threshold

    @public
    def verify(self, a: Bigint, b: Bigint, flag: bool):
        a_above_threshold: bool = a > self.threshold
        b_above_threshold: bool = b > self.threshold
        both_above: bool = a_above_threshold and b_above_threshold
        either_above: bool = a_above_threshold or b_above_threshold
        not_flag: bool = not flag
        assert_(both_above or (either_above and not_flag))

File: conformance/tests/boolean-logic/boolean-logic.runar.rb
require 'runar'

class BooleanLogic < Runar::SmartContract
  prop :threshold, Bigint

  def initialize(threshold)
    super(threshold)
    @threshold = threshold
  end

  runar_public a: Bigint, b: Bigint, flag: Boolean
  def verify(a, b, flag)
    a_above_threshold = a > @threshold
    b_above_threshold = b > @threshold
    both_above = a_above_threshold && b_above_threshold
    either_above = a_above_threshold || b_above_threshold
    not_flag = !flag
    assert both_above || (either_above && not_flag)
  end
end
